runner/shape.go

```diff
@@ -267,6 +267,13 @@ func shapeMerge(a Shape, b Shape) Shape {
 		Logln(`Missing type equality condition for %s merge: [%#v] and [%#v].`, b.Kind, a, b)
 	}
 
+	// Both scalars of same type; nothing to merge here
```

Comment:
I think this should be moved into addUniqueVaried since that function is called in a few places and this non-varied merge should be handled in all cases it can happen.

https://github.com/multiprocessio/datastation/pull/205/files#diff-73a844042e76efc8c400e2aeb4d6ebfa6f323f453bcff9e0e0a204592ee67d5dR157

Reply:
Thanks for your feedback; I was preparing the change you suggested, but noticed that in all _other_ places that call `addUniqueVaried` besides the one above, there's actually no chance that two Scalars could ever be passed in:

- [here](https://github.com/multiprocessio/datastation/blob/main/runner/shape.go#L232) and [here](https://github.com/multiprocessio/datastation/blob/main/runner/shape.go#L239) will always have one `NullShape`
- [here](https://github.com/multiprocessio/datastation/blob/main/runner/shape.go#L208) and [here](https://github.com/multiprocessio/datastation/blob/main/runner/shape.go#L213) will always have one `VariedShape`

I don't think that it'd be wrong to move the check for identical Scalars into `addUniqueVaried` (all tests still pass!), but I feel like that change would blur the semantics of the function and somewhat reduce its readability: it'd have two different responsibilities now, and it would need a new name/signature to reflect that.

Still happy to do the requested change if you'd prefer that. Let me know what you think! Thanks! 🙌

Reply:
Gotcha, makes sense! Thanks for checking into it!